projects/fw_103/src/main.c

```diff
@@ -0,0 +1,68 @@
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+// A6 input, B2 input, log output
+
+// callback func, takes in the channel and reads ADC data from channel into local var
+// then logs the var data
+static void prv_btn_interrupt_handler(const GpioAddress *address, void *context) {
+  AdcChannel *in_channel = (AdcChannel *)context;
+  uint16_t in_data = 0;
+
+  adc_read_raw(*in_channel, &in_data);
```

Comment:
The homework does say to use an adc *converted* reading, rather than a raw read like this. The difference is that converted readings with `adc_read_converted` are scaled so that the unit is millivolts, whereas the raw readings are just whatever garbage came out of the ADC.

Incidentally, there is a new API in `adc.h` that takes GpioAddresses instead of AdcChannels, look in adc.h if you're interested. You don't need to use it for this homework, but an FYI for the future.